src/lib.rs

```diff
@@ -133,6 +133,12 @@ pub trait CustomizeConnection<C, E>: fmt::Debug + Send + Sync + 'static {
     fn on_acquire(&self, conn: &mut C) -> Result<(), E> {
         Ok(())
     }
+
+    /// Called with connections before they are dropped from the pool
+    ///
+    /// The default implementation does nothing
+    #[allow(unused_variables)]
+    fn on_release(&self, conn: &mut C) {}
```

Comment:
Or rather not. There's a comment in `reap_connections()` which say connections shouldn't be dropped while a certain lock is alive. Having `drop_conn()` take the connection by value would violate that.